src/profile-logic/comparison.js

```diff
@@ -0,0 +1,60 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+
+import type { IndexIntoCategoryList, CategoryList } from '../types/profile';
+
+type TranslationMapForCategories = Map<
+  IndexIntoCategoryList,
+  IndexIntoCategoryList
+>;
+export function mergeCategories(
+  categories1: CategoryList,
+  categories2: CategoryList
+): {
+  categories: CategoryList,
+  translationMaps: TranslationMapForCategories[],
+} {
+  const newCategories = [];
+  const translationMaps = [];
+
+  [categories1, categories2].forEach(categories => {
+    const translationMap = new Map();
+    translationMaps.push(translationMap);
+    const insertedCategories: Map<string, IndexIntoCategoryList> = new Map();
+
+    categories.forEach((category, i) => {
+      const { name } = category;
+      const insertedCategoryIndex = insertedCategories.get(name);
+      if (insertedCategoryIndex !== undefined) {
+        translationMap.set(i, insertedCategoryIndex);
+        return;
+      }
+
+      translationMap.set(i, newCategories.length);
+      insertedCategories.set(name, newCategories.length);
+      newCategories.push(category);
+    });
+  });
+
+  return { categories: newCategories, translationMaps };
+}
+
+export function adjustCategories(
```

Comment:
It's not immediately obvious to me why this should be two different functions.

Reply:
If we wanted to do it in one function only, we'd need to pass the full profiles. I think this has some drawbacks:
* we pass more things that we're really interested in in this function
* should we mutate the profiles or return new objects? By handling only the necessary structure we can return new objects but the caller can decide to mutate the profile if that's useful.

I find it easier to reason with when using 2 separate functions.

Reply:
Thanks for the explanation.